ml-agents/mlagents/trainers/components/reward_signals/gail/signal.py

```diff
@@ -42,6 +42,46 @@ def __init__(
             policy, 128, learning_rate, encoding_size, use_actions, use_vail
         )
         _, self.demonstration_buffer = demo_to_buffer(demo_path, policy.sequence_length)
+
+        # check number of visual observations in demmonstration match
+        for i in range(self.policy.vis_obs_size):
```

Comment:
I think we need to check the sizes of the visual obs and vector obs too.

Reply:
Vector obs are checked below. I can check the size of the visual too.